### PR TITLE
Rename `canEditCollective` to `isAdminOfCollectiveOrHost`

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/5926

### CODE REVIEW

File: lib/collective.lib.js [hunk 1]
```diff
@@ -59,8 +59,8 @@ export const expenseSubmissionAllowed = (collective, user) => {
   if (!collective?.settings?.disablePublicExpenseSubmission) {
     return true;
   }
-  // NOTE: canEditCollective curently returns true if ADMIN or HOST admin
-  if (user?.isRoot() || user?.canEditCollective(collective)) {
+  // NOTE: isAdminOfCollectiveOrHost curently returns true if ADMIN or HOST admin
```
Comment (Betree) [inline lib/collective.lib.js:62]: This is now explicit since the function is called `isAdminOfCollectiveOrHost`

```suggestion
```

**Review by Betree — COMMENTED** [review]
Works for me! Can we do an announcement on Slack's engineering channel and give some time before merging to make sure that folks have an opportunity to take this into consideration if they're currently working on a feature with this method?

**Comment by SudharakaP** [comment]
> Works for me! Can we do an announcement on Slack's engineering channel and give some time before merging to make sure that folks have an opportunity to take this into consideration if they're currently working on a feature with this method?

Done at, https://opencollective.slack.com/archives/C0RMV6F8C/p1663148729595839. Thanks for pointing this out. 👍🏽

**Review by Betree — APPROVED** [review]
Good for me to merge, I think the engineering team is now properly informed. Maybe, after merging, do a quick search for `canEditCollective` just to make sure no other instances have been introduced since.

**Pull request merged**